File: C++/Medium/Accepted/removing-stars-from-a-string_2.cpp
class Solution {
public:
    string removeStars(string s) {
        
        int n = s.size();
        stack<char> stk;
        for (int i = 0; i < n; ++i) {
            if (s[i] == '*') {
                while (!stk.empty() && stk.top() == '*') stk.pop();
                if (!stk.empty()) stk.pop();
            } else {
                stk.push(s[i]);
            }
        }
        string res;
        while (!stk.empty()) {
            res.push_back(stk.top());
            stk.pop();
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

File: Rust/Medium/Accepted/removing-stars-from-a-string_2.rs
impl Solution {
    pub fn remove_stars(s: String) -> String {
        
        let mut v = Vec::new();
        for c in s.chars() {
            if c == '*' {
                v.pop();
            } else {
                v.push(c);
            }
        }
        v.into_iter().collect()
    }
}